### PR TITLE
Tutorial CUI Wallet

### PR DESCRIPTION
**Pull request by daiki44**
- [ ] explain the databases
- [ ] put a gif image from deposit to exit

### CODE REVIEW

File: tutorial/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Tutorial [framework name] CUI wallet
+
+In this tutorial, we will implement the CUI Plasma Wallet using [framework name].
+This is the process from Deposit to Exit, which is the sequence of steps required to exchange on Plasma.
+
+## Prerequistes
+
+- Python2.7
```
Comment (tkmct) [inline tutorial/README.md:8]: can you write why python is needed?

File: tutorial/deposit.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
+# 3. Deposit
+
+In this chapter, we will implement deposit to use tokens on Plasma.
+
+## 3-1. Implement deposit
+
+You can call the `deposit` method from plasma light client.
```
Comment (tkmct) [inline tutorial/deposit.md:7]: can you link to API reference of the method?

File: tutorial/README.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# Tutorial [framework name] CUI wallet
+
+In this tutorial, we will implement the CUI Plasma Wallet using [framework name].
+This is the process from Deposit to Exit, which is the sequence of steps required to exchange on Plasma.
```
Comment (tkmct) [inline tutorial/README.md:4]: I think it's better to explain a little bit more in detail about what we will create in this tutorial.

File: tutorial/deposit.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
+# 3. Deposit
+
+In this chapter, we will implement deposit to use tokens on Plasma.
```
Comment (tkmct) [inline tutorial/deposit.md:3]: what about adding little bit more explanation of deposit?
`Deposit will store your main chain token e.g. ETH or ERC20 tokens in depositContract and mint corresponding tokens that you can use in Plasma.`

File: tutorial/deposit.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+      case "deposit":
+        await deposit(client, args[0]);
+        cuiWalletReadLine(client);
+        break;
+      default:
+        console.log(`${command} is not found`);
+        cuiWalletReadLine(client);
+    }
+  });
+}
```
Comment (tkmct) [inline tutorial/deposit.md:31]: can you write explanation of this part? why we need this?

File: tutorial/deposit.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+## 3-1. Implement deposit
+
+You can call the `deposit` method from plasma light client.
+
+```javascript
+const DEPOSIT_CONTRACT_ADDRESS = config.payoutContracts.DepositContract;
+
+async function deposit(client, amount) {
+  console.log("deposit:", amount);
+  await client.deposit(amount, DEPOSIT_CONTRACT_ADDRESS);
```
Comment (tkmct) [inline tutorial/deposit.md:14]: here, we want to stress this simplicity of method call.

File: tutorial/show-balance.md [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+        break;
+      default:
+        console.log(`${command} is not found`);
+        cuiWalletReadLine(client);
+    }
+  });
+}
+```
+
+## 4-2. Check your l2 balance
```
Comment (syuhei176) [inline tutorial/show-balance.md:43]: L2 balance?

File: tutorial/exit.md [hunk 8]
```diff
@@ -0,0 +1,229 @@
...
+  const client = await startLightClient();
+  cuiWalletReadLine(client);
+}
+
+main();
+```
+
+## Tutorial [framework name] CUI Wallet - The End
+
+This tutorial is very simple implementation. This wallet is not near production quality we need a lot more work to make it ready for the Mainnet usage.
```
Comment (syuhei176) [inline tutorial/exit.md:224]: Does it need period?

> This wallet is not near production quality"."

File: tutorial/cui-wallet/README.md [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+
+## Prerequistes
+
+- Python (for `node-gyp`)
+- Node v10.x
+- Docker
+
+## Table of Contents
+
+1. [Setup](/tutorial/setup.md)
```
Comment (syuhei176) [inline tutorial/cui-wallet/README.md:23]: also link was changed.

`/tutorial/cui-wallet/setup.md`

File: tutorial/cui-wallet/setup.md [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+Install the libraries from npm required to implement Plasma Wallet.
+
+After that, copy `config.local.json` file to your application repository root.
+
+```bash
+$ mkdir plasma-wallet
+$ cd plasma-wallet
+$ npm init
+$ npm i ethers
+$ npm i @cryptoeconomicslab/eth-plasma-light-client @cryptoeconomicslab/primitives @cryptoeconomicslab/level-kvs
```
Comment (tkmct) [inline tutorial/cui-wallet/setup.md:39]: ここやっぱprimitives入れないといけないの微妙だな

File: tutorial/cui-wallet/README.md [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+
+Once this tutorial is complete, you will be able to create a Wallet very easily that can do the following.
+
+- deposit Ether and ERC20 tokens to Plasma
+- transfer tokens on Plasma
+- withdraw a token from Plasma to Layer1
+
+> **Notice**
+>
+> This tutorial is very simple implementation. This wallet is not near production quality. We need a lot more work to make it ready for the Mainnet usage.
```
Comment (tkmct) [inline tutorial/cui-wallet/README.md:13]: This wallet is not for production use. **DO NOT** use in Mainnet.

File: getting-started/try-plasma-chamber-in-local.md [hunk 12]
```diff
@@ -1,12 +1,14 @@
 # Try [framework name] in local
+
 Quick start of [framework name] on your laptop.
 You can create a secure and scalable Plasma application by reading this document.
 This quick start using Typescript SDK, you need the latest version of Node.js.
```
Comment (tkmct) [inline getting-started/try-plasma-chamber-in-local.md:5]: actually, we don't need *the latest version* of Node.js. We use v10.

File: getting-started/try-plasma-chamber-in-local.md [hunk 13]
```diff
@@ -18,15 +20,17 @@ Get [plasma-aggregator](https://github.com/cryptoeconomicslab/wakkanay-plasma-ag
...
 ```
+
 Contracts config file `config.local.json` is generated at the root directory after `npm run docker:build` if absent.
 
 ## Create your application
 
 ### 1. Install
 
-To start Plasma transfer, we first need to install Plasma libraries from npm.  
+To start Plasma transfer, we first need to install Plasma libraries from npm.
```
Comment (tkmct) [inline getting-started/try-plasma-chamber-in-local.md:33]: what about like this and explain a little bit about each of the libraries.
```
Let's start by installing [framework name] libraries. You will need following libraries.
-  @cryptoeconomicslab/eth-plasma-light-client: plasma light client implementation for Ethereum
- ...
```

File: tutorial/cui-wallet/exit.md [hunk 14]
```diff
@@ -0,0 +1,274 @@
+# 6. Exit
+
+In this chapter, we will implement exit to withdraw your tokens from Plasma.
```
Comment (yuriko627) [inline tutorial/cui-wallet/exit.md:3]: In this chapter, we will implement exit functionality to withdraw your tokens from Plasma.

File: tutorial/cui-wallet/exit.md [hunk 15]
```diff
@@ -0,0 +1,274 @@
...
+    }
+  });
+}
+```
+
+## 6-5. Exit ether
+
+Launch the CUI Wallet and start the exit process!
+
+Start the app with node command and try typing `exit <amount>`.
```
Comment (yuriko627) [inline tutorial/cui-wallet/exit.md:91]: Start the app with node command and try `exit <amount>`.

File: tutorial/cui-wallet/exit.md [hunk 16]
```diff
@@ -0,0 +1,274 @@
...
+
+```
+$ node app.js
+>> exit 10
+```
+
+## 6-6. Check your exit list
+
+Let's show the Exit List to see if the Exit has been claimed correctly.
```
Comment (yuriko627) [inline tutorial/cui-wallet/exit.md:100]: Let's take a look at exitList to see if the exit has been claimed correctly.

File: tutorial/cui-wallet/exit.md [hunk 17]
```diff
@@ -0,0 +1,274 @@
...
+```
+
+</details>
+
+## Tutorial [framework name] CUI Wallet - The End
+
+Congratulations!
+Now we've developed a CUI Plasma Wallet with basic functions from deposit to exit!
+
+Once again, this tutorial is very simple implementation. This wallet is not near production quality. We need a lot more work to make it ready for the Mainnet usage.
```
Comment (yuriko627) [inline tutorial/cui-wallet/exit.md:267]: Once again, this tutorial and the framework is not production-ready yet. Please refrain from using it on the main net until we announce the alpha version and it is ready for production use.

File: tutorial/cui-wallet/exit.md [hunk 18]
```diff
@@ -0,0 +1,274 @@
...
+
+By the way, we really appriciate your interest in [framework name].
+Looking forward to seeing your next project running on [framework name].
```
Comment (yuriko627) [inline tutorial/cui-wallet/exit.md:270]: We really appreciate your interest in [framework name] from the early stage and look forward to seeing your next project built with [framework name].

**Pull request merged**